### ISSUE TITLE
Jetpack Dashboard as an author is missing

### ISSUE THREAD

**Issue by enejb** [Automattic/jetpack#5519]
Currently running the latest master and logged in as an author. 

This is what I see:
<img width="828" alt="screen shot 2016-11-04 at 12 19 11" src="https://cloud.githubusercontent.com/assets/115071/20019346/214f4ea0-a289-11e6-8563-c27de5f1546e.png">

#### What I expected
I expected to see something telling me how Jetpack was relevant to me as an site author.

**Comment by dereksmart** [comment]
Do you have PBE, ATD, or Publicize enabled? Those are the only modules that are relevant to authors.

**Comment by dereksmart** [comment]
Also if you have stats module enabled, an admin has to opt in to allow authors to be able to see them.  If you do that, there should be stats shown in the dashboard.

**Comment by brbrr** [comment]
@enejb Is this something that you want to be addressed?

**Comment by enejb** [comment]
Yes I think we should rethink about what the interface from the author prospective looks like. As well as other roles. 

For example Protect doesn't seem so relevant here. but some sort of site activity that the user did might be more interesting. 

cc: @MichaelArestad for other suggestions here.
